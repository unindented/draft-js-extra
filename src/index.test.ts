import * as React from "react";
import * as renderer from "react-test-renderer";
import { contentStateToReact, elementsToReact, contentStateToElements, Element } from ".";

describe("draft", () => {
  describe(".contentStateToReact", () => {
    it("converts content state to React nodes", () => {
      const fixture = require("./__fixtures__/empty.json");
      const tree = renderer.create(
        React.createElement("div", undefined, contentStateToReact(fixture)),
      );
      expect(tree.toJSON()).toMatchSnapshot();
    });
  });

  describe(".elementsToReact", () => {
    it("handles strings", () => {
      const elements = ["foo"];
      const tree = renderer.create(
        React.createElement("div", undefined, elementsToReact(elements)),
      );
      expect(tree.toJSON()).toMatchSnapshot();
    });

    it("handles elements", () => {
      const elements = [{ tag: "br" }] as Element[];
      const tree = renderer.create(
        React.createElement("div", undefined, elementsToReact(elements)),
      );
      expect(tree.toJSON()).toMatchSnapshot();
    });

    it("handles trees of elements", () => {
      const elements = [
        {
          tag: "p",
          children: [
            "This is a paragraph with ",
            { tag: "b", children: ["bold"] },
            " and ",
            { tag: "i", children: ["italic"] },
            ".",
          ],
        },
      ] as Element[];
      const tree = renderer.create(
        React.createElement("div", undefined, elementsToReact(elements)),
      );
      expect(tree.toJSON()).toMatchSnapshot();
    });
  });

  describe(".contentStateToElements", () => {
    it("handles an empty block", () => {
      const fixture = require("./__fixtures__/empty.json");
      expect(contentStateToElements(fixture)).toMatchSnapshot();
    });

    it("handles headers", () => {
      const fixture = require("./__fixtures__/headers.json");
      expect(contentStateToElements(fixture)).toMatchSnapshot();
    });

    it("handles unordered lists", () => {
      const fixture = require("./__fixtures__/unordered_list.json");
      expect(contentStateToElements(fixture)).toMatchSnapshot();
    });

    it("handles nested unordered lists", () => {
      const fixture = require("./__fixtures__/unordered_list_nested.json");
      expect(contentStateToElements(fixture)).toMatchSnapshot();
    });

    it("handles ordered lists", () => {
      const fixture = require("./__fixtures__/ordered_list.json");
      expect(contentStateToElements(fixture)).toMatchSnapshot();
    });

    it("handles nested ordered lists", () => {
      const fixture = require("./__fixtures__/ordered_list_nested.json");
      expect(contentStateToElements(fixture)).toMatchSnapshot();
    });

    it("handles blockquotes", () => {
      const fixture = require("./__fixtures__/blockquote.json");
      expect(contentStateToElements(fixture)).toMatchSnapshot();
    });

    it("handles code blocks", () => {
      const fixture = require("./__fixtures__/code_block.json");
      expect(contentStateToElements(fixture)).toMatchSnapshot();
    });

    it("handles paragraphs", () => {
      const fixture = require("./__fixtures__/paragraph.json");
      expect(contentStateToElements(fixture)).toMatchSnapshot();
    });

    it("handles paragraphs with inline styles", () => {
      const fixture = require("./__fixtures__/paragraph_styles.json");
      expect(contentStateToElements(fixture)).toMatchSnapshot();
    });

    it("handles paragraphs with overlapping inline styles", () => {
      const fixture = require("./__fixtures__/paragraph_styles_overlapping.json");
      expect(contentStateToElements(fixture)).toMatchSnapshot();
    });
  });
});
