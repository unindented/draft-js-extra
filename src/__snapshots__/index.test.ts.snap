// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`draft .contentStateToElements handles an empty block 1`] = `
Array [
  Object {
    "children": Array [
      Object {
        "children": Array [],
        "tag": "br",
      },
    ],
    "tag": "p",
  },
]
`;

exports[`draft .contentStateToElements handles blockquotes 1`] = `
Array [
  Object {
    "children": Array [
      "This is a quote:",
    ],
    "tag": "p",
  },
  Object {
    "children": Array [
      "Education is a system of imposed ignorance.",
    ],
    "tag": "blockquote",
  },
]
`;

exports[`draft .contentStateToElements handles code blocks 1`] = `
Array [
  Object {
    "children": Array [
      Object {
        "children": Array [
          "for (var i = 0; i < 10; i++) {
  console.log('OHAI');
}",
        ],
        "tag": "code",
      },
    ],
    "tag": "pre",
  },
]
`;

exports[`draft .contentStateToElements handles headers 1`] = `
Array [
  Object {
    "children": Array [
      "This is a large header!",
    ],
    "tag": "h1",
  },
  Object {
    "children": Array [
      "This is a medium header!",
    ],
    "tag": "h2",
  },
  Object {
    "children": Array [
      "This is a small header!",
    ],
    "tag": "h3",
  },
  Object {
    "children": Array [
      "This is a paragraph.",
    ],
    "tag": "p",
  },
]
`;

exports[`draft .contentStateToElements handles nested ordered lists 1`] = `
Array [
  Object {
    "children": Array [
      "This is a list:",
    ],
    "tag": "p",
  },
  Object {
    "children": Array [
      Object {
        "children": Array [
          "Some item",
          Object {
            "children": Array [
              Object {
                "children": Array [
                  "Some nested item",
                ],
                "tag": "li",
              },
              Object {
                "children": Array [
                  "Some other nested item",
                  Object {
                    "children": Array [
                      Object {
                        "children": Array [
                          "Some deeply nested item",
                        ],
                        "tag": "li",
                      },
                    ],
                    "tag": "ol",
                  },
                ],
                "tag": "li",
              },
            ],
            "tag": "ol",
          },
        ],
        "tag": "li",
      },
      Object {
        "children": Array [
          "Some other item",
        ],
        "tag": "li",
      },
    ],
    "tag": "ol",
  },
]
`;

exports[`draft .contentStateToElements handles nested unordered lists 1`] = `
Array [
  Object {
    "children": Array [
      "This is a list:",
    ],
    "tag": "p",
  },
  Object {
    "children": Array [
      Object {
        "children": Array [
          "Some item",
          Object {
            "children": Array [
              Object {
                "children": Array [
                  "Some nested item",
                ],
                "tag": "li",
              },
              Object {
                "children": Array [
                  "Some other nested item",
                  Object {
                    "children": Array [
                      Object {
                        "children": Array [
                          "Some deeply nested item",
                        ],
                        "tag": "li",
                      },
                    ],
                    "tag": "ul",
                  },
                ],
                "tag": "li",
              },
            ],
            "tag": "ul",
          },
        ],
        "tag": "li",
      },
      Object {
        "children": Array [
          "Some other item",
        ],
        "tag": "li",
      },
    ],
    "tag": "ul",
  },
]
`;

exports[`draft .contentStateToElements handles ordered lists 1`] = `
Array [
  Object {
    "children": Array [
      Object {
        "children": Array [
          "First item",
        ],
        "tag": "li",
      },
      Object {
        "children": Array [
          "Second item",
        ],
        "tag": "li",
      },
    ],
    "tag": "ol",
  },
]
`;

exports[`draft .contentStateToElements handles paragraphs 1`] = `
Array [
  Object {
    "children": Array [
      "This is a paragraph.",
    ],
    "tag": "p",
  },
]
`;

exports[`draft .contentStateToElements handles paragraphs with inline styles 1`] = `
Array [
  Object {
    "children": Array [
      "This is a paragraph with ",
      Object {
        "children": Array [
          "bold",
        ],
        "tag": "b",
      },
      " and ",
      Object {
        "children": Array [
          "italic",
        ],
        "tag": "i",
      },
      ".",
    ],
    "tag": "p",
  },
]
`;

exports[`draft .contentStateToElements handles paragraphs with overlapping inline styles 1`] = `
Array [
  Object {
    "children": Array [
      "This is a paragraph with overlapping ",
      Object {
        "children": Array [
          "bold ",
        ],
        "tag": "b",
      },
      Object {
        "children": Array [
          Object {
            "children": Array [
              "and",
            ],
            "tag": "i",
          },
        ],
        "tag": "b",
      },
      Object {
        "children": Array [
          " italic",
        ],
        "tag": "i",
      },
      ".",
    ],
    "tag": "p",
  },
]
`;

exports[`draft .contentStateToElements handles unordered lists 1`] = `
Array [
  Object {
    "children": Array [
      Object {
        "children": Array [
          "Some item",
        ],
        "tag": "li",
      },
      Object {
        "children": Array [
          "Some other item",
        ],
        "tag": "li",
      },
    ],
    "tag": "ul",
  },
]
`;

exports[`draft .contentStateToReact converts content state to React nodes 1`] = `
<div>
  <p>
    <br />
  </p>
</div>
`;

exports[`draft .elementsToReact handles elements 1`] = `
<div>
  <br />
</div>
`;

exports[`draft .elementsToReact handles strings 1`] = `
<div>
  foo
</div>
`;

exports[`draft .elementsToReact handles trees of elements 1`] = `
<div>
  <p>
    This is a paragraph with 
    <b>
      bold
    </b>
     and 
    <i>
      italic
    </i>
    .
  </p>
</div>
`;
